### PR TITLE
Address chapter 6 feedback

### PR DESCRIPTION
**Pull request by chrishtr**
* Adds two go-further blocks to other chapters based on chapter 6 feedback
* Renames `Layout.get_font` to `font` to avoid a conflict with the `get_font` global function

### CODE REVIEW

File: book/scripts.md [hunk 1]
```diff
@@ -1067,6 +1067,32 @@ def add_entry(params):
     return show_comments()
 ```
 
+::: {.further}
+
+Search engines are constantly [crawling] the web and [indexing] all of the web
+pages they can find. In the early days, indexing was just a matter of loading
+the HTML, parsing it and extracting the information. But these days, a lot of
+[single-page app][spa] sites use JavaScript to ["hydrate"][hydration]
+[^why-hydrate] their site into its full contents. On such sites, before
```
Comment (pavpanchekha) [inline book/scripts.md:1076]: I think the placement of `[^why-hydrate]` here adds an extra space between that and `"hydrate"`, which is not right.

File: src/lab6.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 import tkinter.font
 from lab1 import URL
 from lab2 import WIDTH, HEIGHT
-from lab3 import FONTS, get_font
```
Comment (pavpanchekha) [inline src/lab6.py:14]: Wait, why? I think we want to import the global FONTS cache.

File: src/lab10.py [hunk 3]
```diff
@@ -11,7 +11,6 @@
 import urllib.parse
 import dukpy
 from lab2 import WIDTH, HEIGHT, HSTEP, VSTEP, SCROLL_STEP
-from lab3 import FONTS, get_font
```
Comment (pavpanchekha) [inline src/lab10.py:14]: I think we should keep these imports around—one day I'll fix the inlining system so we don't need them but for now please keep them.

File: src/lab7.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 import tkinter
 import tkinter.font
 from lab2 import WIDTH, HEIGHT, HSTEP, VSTEP, SCROLL_STEP
-from lab3 import FONTS, get_font
```
Comment (pavpanchekha) [inline src/lab7.py:13]: Same on keeping this.

File: src/lab8.py [hunk 5]
```diff
@@ -11,7 +11,7 @@
 import tkinter.font
 import urllib.parse
 from lab2 import WIDTH, HEIGHT, HSTEP, VSTEP, SCROLL_STEP
-from lab3 import FONTS, get_font
```
Comment (pavpanchekha) [inline src/lab8.py:14]: Same on keeping this

File: src/lab9.py [hunk 6]
```diff
@@ -12,7 +12,6 @@
 import urllib.parse
 import dukpy
 from lab2 import WIDTH, HEIGHT, HSTEP, VSTEP, SCROLL_STEP
-from lab3 import FONTS, get_font
```
Comment (pavpanchekha) [inline src/lab9.py:15]: Same on keeping this

**Review by pavpanchekha — CHANGES_REQUESTED** [review]
Please undo the import stuff but other than that looks good. I think renaming `get_font` to just `font` is a great compromise.

File: src/lab10.py (hunk 3 above)
Comment (chrishtr) [reply src/lab10.py:14]: Ok, but can you tell me why we need them for the inliner? I didn't know about this (or forgot).

Comment (chrishtr) [reply src/lab10.py:14]: I left FONTS but not get_font, which I don't think is needed?

**Comment by chrishtr** [comment]
> Please undo the import stuff but other than that looks good. I think renaming `get_font` to just `font` is a great compromise.

Done

**Comment by pavpanchekha** [comment]
Yeah, ok, so the summary of the annoying import rule here is that for the inliner to work properly, we need to import not just any functions / constants that we use (how importing normally works), but also any transitive dependencies. So, for example, the `Layout.font` function uses the `get_font` global function, which in turn uses the `FONTS` global constant, so we need to import them. (And in fact since we keep using the `font` method throughout, we need to keep using importing `get_font` throughout.)

File: src/lab10.py (hunk 3 above)
Comment (pavpanchekha) [reply src/lab10.py:14]: I am guessing that `BlockLayout` still calls its `font` method which in turn still calls `get_font`, which we still need.

**Review by pavpanchekha — CHANGES_REQUESTED** [review]

**Comment by pavpanchekha** [comment]
Honestly it would be better to just fix the damn importer, but I guess we are prioritizing the book pass for now :)

**Comment by chrishtr** [comment]
Updated.

**Review by pavpanchekha — APPROVED** [review]

**Pull request merged**